v1/build_config.go

```diff
@@ -126,7 +126,7 @@ func (c *Client) CreateBuildConfig(user, name string) (*BuildConfig, error) {
 //
 // Actual API: "Create Build Config Version"
 func (c *Client) UploadBuildConfigVersion(v *BuildConfigVersion, metadata map[string]interface{},
-	data io.Reader, size int64) error {
+	vars map[string]string, data io.Reader, size int64) error {
```

Comment:
Agreed that would be better. Would like to do it outside of the scope of this PR though, and for consistency we would make the same breaking change to all other functions.